tests/pipeline/test_blaze.py

```diff
@@ -0,0 +1,456 @@
+"""
+Tests for the blaze interface to the pipeline api.
+"""
+from __future__ import division
+
+from collections import OrderedDict
+from datetime import timedelta
+from itertools import chain
+from unittest import TestCase
+import warnings
+
+import blaze as bz
+from datashape import dshape, var, Record
+import numpy as np
+import pandas as pd
+from pandas.util.testing import assert_frame_equal
+from toolz import keymap
+from toolz.curried import operator as op
+
+from zipline.pipeline import Pipeline, CustomFactor
+from zipline.pipeline.data import DataSet, BoundColumn
+from zipline.pipeline.engine import SimplePipelineEngine
+from zipline.pipeline.loaders.blaze import (
+    from_blaze,
+    BlazeLoader,
+    NoDeltasWarning,
+    NonNumpyField,
+    NonPipelineField,
+)
+from zipline.utils.test_utils import tmp_asset_finder
+
+
+nameof = op.attrgetter('name')
+dtypeof = op.attrgetter('dtype')
+
+
+class BlazeToPipelineTestCase(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.dates = pd.date_range('2014-01-01', '2014-01-03')
+        dates = (
+            [pd.Timestamp('2014-01-01')] * 3 +
+            [pd.Timestamp('2014-01-02')] * 3 +
+            [pd.Timestamp('2014-01-03')] * 3
+        )
+        cls.sids = sids = ord('A'), ord('B'), ord('C')
+        cls.df = df = pd.DataFrame({
+            'sid': sids * 3,
+            'value': tuple(
```

Comment:
I think it's actually fewer characters and clearer to just write `(0, 1, 2, 1, 2, 3, 2, 3, 4)` here.